OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -40,6 +36,9 @@ public class BuildingInfo : ITraitInfo, IOccupySpaceInfo, IPlaceBuildingDecorati
 			"Set to '-1' to disable adjacency checks.")]
 		public readonly int Adjacent = 2;
 
+		[Desc("Type of influence, defined by the `GivesBuildableArea` traits of the nearby actors, this can be build on.")]
+		public readonly string AdjacentType = null;
```

Comment:
Can you please make this a `public readonly HashSet<string> AreaTypes = new HashSet<string>();`?

Two things here:
* The *Type name should match the trait name.
* Supporting multiple here gives more flexibility.  The only change later is to switch the `.Contains(AdjacentType)` with `.Overlaps(AreaTypes)`.